### ISSUE TITLE
getting "ECONNRESET" when using SSL

### ISSUE THREAD

**Issue by declan782** [amqp-node/amqplib#708]
hi there,

I'm connecting a RabbitMQ cluster behind a load balancer with amqplib:
```
var connectObj = {
  protocol: 'amqps',
  hostname: '<hostname>',
  port: 5671,
  username: '<username>',
  password: '<password>',
  vhost: '<vhost>',
};
var opts = {ca:[fs.readFileSync("cacerts.pem")]};// the PEM contains several server certs. 
amqp.connect(connectObj,opts).then(function(conn) {
  process.once('SIGINT', function() { conn.close(); });
  return conn.createChannel().then(function(ch) {
    var ok = ch.assertQueue('test', {durable: false});
    ok = ok.then(function(_qok) {
      return ch.consume('hello', function(msg) {
        console.log(" [x] Received '%s'", msg.content.toString());
      }, {noAck: true});
    });

    return ok.then(function(_consumeOk) {
      console.log(' [*] Waiting for messages. To exit press CTRL+C');
    });
  });
}).catch(console.warn);
```

but keep getting below error:
```
Error: read ECONNRESET
    at TLSWrap.onStreamRead (node:internal/stream_base_commons:217:20) {
  errno: -4077,
  code: 'ECONNRESET',
  syscall: 'read'
}
```
or
```
Error: write ECONNRESET
    at WriteWrap.onWriteComplete [as oncomplete] (node:internal/stream_base_commons:94:16)
    at handleWriteReq (node:internal/stream_base_commons:61:21)
    at writeGeneric (node:internal/stream_base_commons:149:15)
    at Socket._writeGeneric (node:net:917:11)
    at Socket._write (node:net:929:8)
    at writeOrBuffer (node:internal/streams/writable:392:12)
    at _write (node:internal/streams/writable:333:10)
    at Writable.write (node:internal/streams/writable:337:10)
```

any help is appreciated.

**Comment by declan782** [comment]
and I used below command to verify the SSL:
```
openssl s_client -CAfile cacerts.pem -connect <hostname>:5671
```
I can find the result is:
```
Verify return code: 0 (ok)
```

**Comment by cressie176** [comment]
Hi @declan782,

ECONNRESET indicates that the connection was closed by a peer, which could be the RabbitMQ server or your load balancer. 

- Have you checked the RabbitMQ / Load balancer logs (increasing the logging level if there's nothing useful)?
- At what point does the error occur - after a successful connection, or after your script has received the SIGINT?

**Comment by cressie176** [comment]
Also worth checking [previous issues](https://github.com/amqp-node/amqplib/search?q=ECONNRESET+SSL&type=issues) if you haven't done so already.

**Comment by declan782** [comment]
hi @cressie176 ,
Thanks for your reply. 
The error occors when the script is started, NOT after a successful connection.

btw, we have another app, which is written in Java, it's ok to connect with SSL.

**Comment by cressie176** [comment]
I have a [working example](https://github.com/cressie176/amqplib-696) using docker-compose. Maybe it will help.

**Comment by declan782** [comment]
The issue is resolved after upgrading amqplib from 0.5.5 to 0.10.3. NODE version is 16.16.0.
if only newer versions can support SSL, appreciate if it can be stated it out in the guide. Thanks.

**Comment by cressie176** [comment]
Thanks for updating @declan782,

I found the following in the [changelog](https://github.com/amqp-node/amqplib/blob/main/CHANGELOG.md#changes-in-v080) 

> Use hostname as TLS servername, to help with using servers behind load balancers (https://github.com/squaremo/amqp.node/pull/567, thanks to @carlhoerberg and commenters)

https://github.com/amqp-node/amqplib/pull/567

I suspect this may have been what caused the problems in your case
